java/ql/lib/semmle/code/java/CompilationUnit.qll

```diff
@@ -31,5 +31,50 @@ class CompilationUnit extends Element, File {
    */
   Module getModule() { cumodule(this, result) }
 
+  /**
+   * Gets a type which is available by its simple name in this compilation unit.
+   * Reasons for this can be:
+   * - The type is declared in this compilation unit as top-level type
+   * - The type is imported
+   * - The type is declared in the same package as this compilation unit
+   * - The type is declared in the package `java.lang`
+   */
+  ClassOrInterface getATypeAvailableBySimpleName() {
```

Comment:
Maybe `AvailableBySimpleName` -> `InScope`?

Reply:
In that case should the shadowing and simple name aspects be removed from the comments? Because a shadowed type would still be in scope.

Reply:
I don't know off hand if the JLS formally defines "in scope"? In either case though explicitly mentioning this tricky corner seems useful, so I'd keep it.